### ISSUE TITLE
cabin: write a package called "parse-app-info"

### ISSUE THREAD

**Issue by niftylettuce** [niftylettuce/bug-bounties#13]
Similar to [parse-request][] or [parse-err][], it will parse information about an application process.

Use the latest version of `lass` to build this package, `npm install -g lass` and see https://lass.js.org.

Usage should be asynchronous and as follows:

```js
const parseAppInfo = require('parse-app-info');

(async () => {
  const appInfo = await parseAppInfo();
  console.log('appInfo', appInfo);
})();
```

Default object returned as `appInfo` should be:

- `name` - use https://github.com/sindresorhus/read-pkg-up#readme to parse `name` from `package.json`
- `version` - use https://github.com/sindresorhus/read-pkg-up#readme to parse `version` from `package.json`
- `node` - Node.js version using `process.version`
- `hash` - long git commit hash using https://github.com/node-modules/last-commit-log
- `tag` - git tag using https://github.com/node-modules/last-commit-log
- `environment` - either `process.env.NODE_ENV || 'development'`
- `hostname` - either `process.env.HOSTNAME || require('os').hostname()`
- `pid` - current `process.pid`

There should also be tests written for this.

[parse-request]: https://github.com/niftylettuce/parse-request
[parse-err]: https://github.com/niftylettuce/parse-err

**Comment by pke** [comment]
@niftylettuce there you go https://github.com/pke/parse-app-info

Reaching 100% code coverage with ENV vars is not easy, so I excluded those paths from the coverage.

**Comment by niftylettuce** [comment]
taking a look @pke!

curious - how is your experience using lass so far?  any issues?

**Comment by niftylettuce** [comment]
@pke this looks fantastic.  can you handle these minor nits? https://github.com/pke/parse-app-info/issues/1

once done I will send over bounty! 👍 ❤️

**Comment by pke** [comment]
re lass. Works good, I have a few suggestions though:

1. Missing git repo 

Opening the pages for travis, coverage and github all will show a 404 if the repository was not created first at github. 
Maybe this could be done via the github API during the lass project creation process. However that also would have to ask the user for her GH password which some might be reluctant to give. Creating the repo *before* running helps.
Instead maybe you could do a quick check when you ask for the repo URL (which you nicely generate already) and if its a 404 then remind the user to create the repo now as an empty repo.

2. git message linting

The commit linting was harsh at first cause I followed your commit messages which do no adhere to the required "type(..): message" format. However I solved that quickly since I am also using this format in my own projects more and more.

3. Travis CI badge URL in readme should be updated

The travis-ci.org is deprectated and new projects should use the free tier for open source projects on travis-ci.com (https://blog.travis-ci.com/2018-05-02-open-source-projects-on-travis-ci-com-with-github-apps)
The opened travis CI tab in the browser for the project should also be travis-ci.com

Thats all ;) Good work!

**Comment by niftylettuce** [comment]
I have left remaining comments at <https://github.com/pke/parse-app-info/issues/1>.

Also sent your bug bounty reward/payout over PayPal just now.

I've fixed issues you described in Lass and published new version v1.0.8.

You can install and try now with `npm install -g lass@latest`.  Also you can confirm proper version installed with `lass --version`.
